pkg/forklift-api/webhooks/validating-webhook/admitters/plan-admitter.go

```diff
@@ -7,93 +7,60 @@ import (
 
 	"encoding/json"
 	"fmt"
+	"net/http"
+
 	admissionv1 "k8s.io/api/admission/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
-	"net/http"
 
 	"github.com/konveyor/forklift-controller/pkg/apis"
 	api "github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	liberr "github.com/konveyor/forklift-controller/pkg/lib/error"
 	"github.com/konveyor/forklift-controller/pkg/forklift-api/webhooks/util"
 )
 
 type PlanAdmitter struct {
+	client client.Client
+	plan api.Plan
+	sourceProvider api.Provider
+	destinationProvider api.Provider
 }
 
-func (admitter *PlanAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
-	log.Info("Plan admitter was called")
-	raw := ar.Request.Object.Raw
-
-	plan := &api.Plan{}
-	err := json.Unmarshal(raw, plan)
-	if err != nil {
-		return util.ToAdmissionResponseError(err)
-	}
+func (admitter *PlanAdmitter) validateStorage() *admissionv1.AdmissionResponse {
```

Comment:
the refactoring is nice!
nitpicking - how about returning an `error` instead of `*admissionv1.AdmissionResponse` and in the callers to do
```golang
if err != nil {
	return util.ToAdmissionResponseError(err)
}
```
as we never use the allows-response